### PR TITLE
feat: Add support for aws_autoscaling_policy

### PR DESCRIPTION
**Pull request by Sudokamikaze**
## Description
This PR adds support for aws_autoscaling_policy resource

Support is fully complete and covers all settings mentioned in documentation(https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_policy)

## Motivation and Context
I decided to implement support for this resource, as it's a part of ASGs, brings (important in my opinion) feature to scale-in / scale out based on CPU load metric (predefined_metric_type = ASGAverageCPUUtilization)

## How Has This Been Tested?
- [ *] I have tested and validated these changes using one or more of the provided `examples/*` projects
I tested this PR creating ASG with addition of following block of code:
```
  create_scaling_policy = true
  scaling_policies = {
    was-avg-cpu-policy-greater-than-xx = {
      policy_type               = "TargetTrackingScaling"
      estimated_instance_warmup = 1200
      target_tracking_configuration = {
        predefined_metric_specification = {
          predefined_metric_type = "ASGAverageCPUUtilization"
        }
        target_value = 50.0
      }
    }
  }
```
In `terraform plan` I see following result:
```
  # module.workers-ami-test.module.autoscaling.aws_autoscaling_policy.this["was-avg-cpu-policy-greater-than-xx"] will be created"aws_autoscaling_policy" "this" {
      + arn                       = (known after apply)
      + autoscaling_group_name    = "workers-ami-test"
      + estimated_instance_warmup = 1200
      + id                        = (known after apply)
      + metric_aggregation_type   = (known after apply)
      + name                      = "was-avg-cpu-policy-greater-than-xx"
      + policy_type               = "TargetTrackingScaling"

      + target_tracking_configuration {
          + disable_scale_in = false
          + target_value     = 50

          + predefined_metric_specification {
              + predefined_metric_type = "ASGAverageCPUUtilization"
            }
        }
    }
```
I also made a test without it, as variables have default value of `false` and `null`, it does not break anything

### CODE REVIEW

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Good start. Please update the code in the `examples` and docs in README.

**Comment by Sudokamikaze** [comment]
> Good start. Please update the code in the `examples` and docs in README.

Agreed, I'll update them and will let you know once it's done, thank you :)

**Comment by Qanop** [comment]
Great message! I was just looking for this solution today while modifying the code.

**Comment by Sudokamikaze** [comment]
@antonbabenko I've added example based on average CPU load and added missing outputs

File: outputs.tf
Comment (bryantbiggs) [inline outputs.tf:118]:
```suggestion
# Autoscaling Policy
```

File: main.tf
Comment (bryantbiggs) [inline main.tf:477]:
```suggestion
# Autoscaling Policy
```

File: main.tf
Comment (bryantbiggs) [inline main.tf:480]: I haven't tested this and it may work, but I suspect we might have to do:

```suggestion
  for_each = { for k,v in var.scaling_policies : k => v if var.create_asg && var.create_scaling_policy }
```

File: main.tf
Comment (bryantbiggs) [inline main.tf:482]: this looks great and I completely agree, but users can sometimes be thrown off by this. instead we can give them a little help by overloading or adding a fallback (support two ways)

```suggestion
  name                      = lookup(each.value, "name", each.key)
```

so they can specify the name in the map as per normal, but if not then we can fall back to use the map key

**Comment by Sudokamikaze** [comment]
@bryantbiggs Hey, TYVM for the review & changes

I've implemented them, slightly tested by `plan` & updated documentation slightly

File: main.tf
Comment (antonbabenko) [inline main.tf:551]: Both `predefined_metric_type` and `resource_label` are required.

**Review by antonbabenko — CHANGES_REQUESTED** [review]
LGTM (almost)

File: variables.tf
Comment (antonbabenko) [reply variables.tf:544]: Then they don't have to specify values in `scaling_policies` if they don't want to make them.

Comment (antonbabenko) [reply variables.tf:544]: The same logic as with schedules (above)

Comment (Sudokamikaze) [reply variables.tf:544]: Makes sense, will implement all needed changes, thank you :)

File: outputs.tf
Comment (antonbabenko) [inline outputs.tf:121]:
```suggestion
output "autoscaling_policy_arns" {
```

File: variables.tf
Comment (antonbabenko) [inline variables.tf:549]:
```suggestion
  type        = any
```

File: variables.tf
Comment (antonbabenko) [inline variables.tf:538]:
```suggestion
# Autoscaling policy
```

**Review by antonbabenko — CHANGES_REQUESTED** [review]
Looks pretty good. There are few remaining comments, also please update outputs in `examples`, and make CI green&passing. Thanks!

**Comment by Sudokamikaze** [comment]
> Looks pretty good. There are few remaining comments, also please update outputs in `examples`, and make CI green&passing. Thanks!

BTW, I'm seeing `First-time contributors need a maintainer to approve running workflows. Learn more.` each time, it could be much better if those linters could run without approval, so I could fix it before asking you to re-review

Also I forgot to mention, in previous review round, I implemented `target_value` as required argument for both blocks (to align more with official documentation)

**Comment by antonbabenko** [comment]
This is how GitHub is working for us. You don't need to wait for GitHub Actions to run after you commit, because you can run `pre-commit run -a` to make all required changes locally before committing, and GitHub Actions will do the same and it will be green afterward. Also, you can run this once - `pre-commit install --install-hooks` and then run `git commit` normally.

I've just run it locally.

Thank you very much for this PR! I am going to merge it now.

**Comment by antonbabenko** [comment]
This PR is included in version 4.11.0 :tada:

**Comment by Sudokamikaze** [comment]
> This is how GitHub is working for us. You don't need to wait for GitHub Actions to run after you commit, because you can run `pre-commit run -a` to make all required changes locally before committing, and GitHub Actions will do the same and it will be green afterward. Also, you can run this once - `pre-commit install --install-hooks` and then run `git commit` normally.
> 
> I've just run it locally.
> 
> Thank you very much for this PR! I am going to merge it now.

Huge thank you! In this PR I learned good practices and looking forward to contribute to your modules, next time it'll be much less to review ;)

**Comment by antonbabenko** [comment]
You are more than welcome! We have plenty of work to do in `terraform-aws-modules`.

**Pull request merged**
